### ISSUE TITLE
Issues with "debloated" Windows

### ISSUE THREAD

**Issue by InvoxiPlayGames** [InvoxiPlayGames/FestivalInstrumentMapper#64]
**This issue only applies if you've ever "de-bloated" Windows, or used any modded "distro" of Windows.** If you aren't using them, or don't know what that means, check for other issues or submit a new one.

Many Windows "de-bloat" scripts or pre-made "distros" end up removing a bunch of random services, drivers and applications without considering what the underlying services actually do and how they interact with one another. These often remove anything with "Xbox" in the name, sometimes including the services needed to support or emulate Xbox One controllers. If you've used one of these scripts, these issues can show up, even if everything is configured properly:

- "Failed to initialise GipSyntheticEx" when starting or "Failed to create synthetic controller" when trying to map a guitar.
- The entire app freezing when trying to map a guitar.
- The app claiming the mapping was successful, but Fortnite doesn't see any guitar.

I'd like to know if anyone has found a solution to recovering Xbox One controller emulation support from the messed up state that Windows is put in. If you've been able to get it working, without a reinstall, please let others know what worked in this thread.

### Known so far:
- "Xbox Accessory Management Service"/XboxGipSvc has to be installed and not disabled.

**Because the issues are so varied, we will still not be providing active help to people using these versions of Windows, this issue is just to help people try to find a solution.**

**Comment by april83c** [comment]
I'm using [AME 11](https://git.ameliorated.info/Styris/AME-11), where XboxGipSvc is still available and running, and I have the issue where Fortnite doesn't see the guitar. It seems that the only Xbox things that AME 11 messes with are

```
# Configuration/tasks/services.yml
  - !service:
    name: "XblAuthManager"
    operation: stop
  - !service:
    name: "XblGameSave"
    operation: stop
  - !service:
    name: "XboxNetApiSvc"
    operation: stop
```

```
# Configuration/tasks/appx.yml
  - !appx: {name: '*Xbox.TCUI*', type: family}
  - !appx: {name: '*XboxApp*', type: family}
  - !appx: {name: '*XboxGameCallableUI*', type: family}
  - !appx: {name: '*XboxGameOverlay*', type: family}
  - !appx: {name: '*XboxGamingOverlay*', type: family}
  - !appx: {name: '*XboxIdenitity*', type: family}
  - !appx: {name: '*XboxSpeechToTextOverlay*', type: family}
```

Enabling the stopped services doesn't fix the issue, so I assume one of the removed Appx packages is required for controller authentication? I'd assume it's probably `Xbox.TCUI` or `XboxGameCallableUI` but I don't really have a way to test (I guess I could use a VM and get rid of them one by one until it breaks but I don't really think it's feasible to run Fortnite in a VM... is there a better way to test if the controller authentication works (than running Fortnite)?)

**Comment by InvoxiPlayGames** [comment]
If you filter Device Manager by "Devices by connection", a new "USB Input Device" listing will show up as a "HID-compliant Device" with the VID/PID of the PDP Jaguar.

![image](https://github.com/InvoxiPlayGames/FestivalInstrumentMapper/assets/22731889/df889181-4c59-4601-aae7-5f4d5126f228)

This doesn't necessarily mean that Fortnite will see and be able to use the guitar (since the API for controller emulation isn't the same as the API for using IGameInput), but it does mean that the emulated controller is at the very least showing up properly.

You can technically "*run*" Fortnite on a VM if you have GPU support (VMware Workstation *should* have good enough DirectX 11?) and replace the anticheat executables (ending _BE, _EAC_EOS) with copies of the full game client, but you won't be able to join any game (you'll get kicked soon after you start a song, and might get banned so I'd use a different account, *I am not responsible etc etc*). It'd be good enough for testing controllers in the menus if you're willing to copy over a 40GB game.

**Comment by SMHUNTERSM** [comment]
My windows version is original and is not modified in this way the services are installed and working but when mapping the controls the app freezes for a while and then reacts to show error I am trying to use the wiiuitar.

**Comment by rodentrob** [comment]
I think I found the solution to this and other related issues I see posted:

**TLDR;**

- Install/Reinstall Microsoft Edge/Edge Enterprise AND Microsoft Edge Webview2
- Make your guitar go in a player slot except Player (1)
- Your PC should play Xbox games online (more on services troubleshoot)


**Tested in:**
_Xplor 360
Stratocaster 360
Les Paul 360 wireless with receiver_

**Troubleshoot:**

- **"This software already installed on your system (Edge/Edge Enterprise/Edge Webview2)"**

Remember to backup your registry
https://answers.microsoft.com/en-us/windows/forum/all/this-software-already-installed-on-your-system/8131f546-8d4a-40a6-a5ac-27c0b224d28d
(I also deleted this registry before installing edge again HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\msedge.exe\0)


- **"I have it installed but it doesnt let me repair it or delete it (Edge/Edge Enterprise/Edge Webview2)"** 

I used this debloater in my last test, ONLY use delete Edge from the tweaks section (you dont need to watch the full video )
Once uninstalled, you can install it via the same prompt again along with webview2 or use the GUI
https://www.youtube.com/watch?v=5_AaHXrelTE


- **"Fortnite getting crazy and camera turning around (HidHide not working)"**

_In my case HidHide never worked so I uninstalled it, here are some workarounds (You need to have installed Edge and webview2 correctly):_

1. First connect another controller, or emulate a controller using x360ce, it should be in the slot of Player (1)
2. Then connect your guitar controller so it appears as Player (2) (or 3,4 doesnt matter as long it isnt 1)
3. Run FestivalInstrumentMapper
4. Refresh devices and select your guitar
5. Make sure the "Tell HidHide to Hide ..." in FestivalInstrumentMapper is disabled (or simply don't install HidHide so it's disabled)
6. Press Start mapping
7. Start Fortnite
8. It should be working now, you can control the menu with the keyboard or the Player 1 controller


- **"I got microsoft bloat again after installing Edge and Webview2":**

In the pic are some of the debloated apps I safely deleted via Bloatbox
Not pictured but also deleted, Paint3D, Calendar and Cortana (Cortana id is F024294D.407397AB99398)
![bloat](https://github.com/InvoxiPlayGames/FestivalInstrumentMapper/assets/169528316/98180c91-c3cc-4c54-b1e8-a1ebf3079216)


- **"What services are running in automatic"**
_In my case:_
XblAuthManager
XblGameSave
XboxGipSvc
XboxNetApiSvc
GamingServices
GameInputService
The Xbox App

**Comment by rodentrob** [comment]
I updated my previous post with more info and better formating

**Comment by SMHUNTERSM** [comment]
@rodentrob But my problem is that when I connect my wiiuitar using the wiitharthing program to recognize it as an xbox guitar (since I don't have the raphnet) when I select my guitar in the festivalmapper program and click on mapping the program stays a few seconds without responding and then returns to respond but with the error 80080005

**Comment by rinrawr** [comment]
Sorry for reviving this old issue but has anyone had the issue where XboxGipSvc is just gone from Windows Services? I've tried so many different things but the Accessory Management service is just gone and I can't find anyway to get it back

**Comment by novoide** [comment]
I fixed it after removing XboxGipSvc from regedit.
Equipo\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\XboxGipSvc\Parameters

**WARNING: DO NOT REMOVE xboxgipsvc.dll FROM SYSTEM32!!!!**
<img width="336" height="90" alt="Image" src="https://github.com/user-attachments/assets/89b9813e-efed-46f7-99de-5e371a97f205" />

After that I rebooted and service disapeared as expected.
Then I installed Xbox Gaming Services and I manually added XboxGipSvc into regedit via PowerShell:

`New-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Services\XboxGipSvc" -Force
Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\XboxGipSvc" -Name "Start" -Value 3 -Type DWord
Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\XboxGipSvc" -Name "Type" -Value 16 -Type DWord
Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\XboxGipSvc" -Name "ErrorControl" -Value 1 -Type DWord
Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\XboxGipSvc" -Name "ImagePath" -Value "C:\Windows\system32\svchost.exe -k netsvcs -p" -Type String
Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\XboxGipSvc" -Name "DisplayName" -Value "Xbox Accessory Management Service" -Type String
Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\XboxGipSvc" -Name "ObjectName" -Value "LocalSystem" -Type String
New-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Services\XboxGipSvc\Parameters" -Force`


After this I added a parameter with dll location.

<img width="1026" height="156" alt="Image" src="https://github.com/user-attachments/assets/875c29e9-0b90-4005-a121-9d74a6363ecf" />

Reboot and Enjoy!

<img width="682" height="636" alt="Image" src="https://github.com/user-attachments/assets/54c2ac41-63c9-4fc2-a165-5a833a2894d1" />
